### PR TITLE
Fix bug 1407532 Update git submodules buildpack, refactor code respon…

### PR DESCRIPTION
**Pull request by jotes**
Hey @mathjazz and @adngdb!
Could you look at this patch?
Thanks in advance.

### CODE REVIEW

File: app.json [hunk 1]
```diff
@@ -32,6 +32,10 @@
      }
    ],
    "env": {
+     "SITE_URL": {
+       "description": "Access url to the instance e.g. https://<app name>.herokuapp.com without a trailing slash.",
+       "required": true
+     },
```
Comment (mathjazz) [inline app.json]: I remember we had issues setting `SITE_URL` during build. Has that changed?

File: app.json [hunk 2]
```diff
@@ -109,7 +113,7 @@
    },
    "buildpacks": [
      {
-       "url": "https://github.com/dmathieu/heroku-buildpack-submodules#b37ffe4361bb9c975dd8e93068c9d296365d748c"
+       "url": "https://github.com/dmathieu/heroku-buildpack-submodules#0caf30af7737bf1bc32b7aafc009f19af3e603c1"
```
Comment (mathjazz) [inline app.json]: Why do we need to change the buildpack? Shall we do that on production, too?

File: runtime.txt [hunk 3]
```diff
@@ -0,0 +1 @@
+python-2.7.14
```
Comment (mathjazz) [inline runtime.txt]: I wonder why do we need to specify this manually now. When deploying to production, I see this:
```
The latest version of Python 2 is python-2.7.14 (you are using python-2.7.12, which is unsupported).
We recommend upgrading by specifying the latest version (python-2.7.14).
```

File: app.json [hunk 4]
```diff
@@ -32,6 +32,10 @@
      }
    ],
    "env": {
+     "SITE_URL": {
+       "description": "Access url to the instance e.g. https://<app name>.herokuapp.com without a trailing slash.",
```
Comment (mathjazz) [inline app.json]: I would be more strict here. Instead of **e.g.**, this **has to be** "https://<app name>.herokuapp.com", right?

**Comment by jotes** [comment]
@mathjazz Hey, I'll update the commit message and I'll try to respond to all your questions inline.

File: runtime.txt (hunk 3 above)
Comment (jotes) [reply runtime.txt]: Unfortunately, I can't add any comments to this file :(
Python3 is the default python for the new apps - that probably doesn't affect existing production but it's breaking deployment of new apps if you don't specify runtime explicitly.
Related docs: https://devcenter.heroku.com/changelog-items/1169

File: app.json (hunk 4 above)
Comment (jotes) [reply app.json]: :+1:

File: app.json (hunk 1 above)
Comment (jotes) [reply app.json]: Previously, Pontoon was able to detect existing domain automatically after deployment.

I took that approach to simplify deployment as much as it was possible.
However, after a few versions of Django (request.get_host() throws an exception if there's no right SITE_URL defined) the process got harder.
The code was already hard to maintain and required of the magic (lazily loaded django settings).
That's why I decided to remove all unnecessary code in the favor of a small requirement from the user.

File: app.json (hunk 2 above)
Comment (jotes) [reply app.json]: I'm not sure about the production - Heroku may work differently for already deployed apps and previous version of this particular buildpack will work perfectly fine.
However, for the new builds https://github.com/dmathieu/heroku-buildpack-submodules/blob/master/bin/install_git_submodules.rb throws an error and https://github.com/dmathieu/heroku-buildpack-submodules/commit/35622728df8757a1bacde155290fcd2960ff1a11 seems to fix that.

Comment (mathjazz) [reply app.json]: Gotcha!

**Comment by mathjazz** [comment]
Thanks for explainig all this! +1 for simplifying the codebase!

The only thing I'd change is that `SITE_URL` description, other than that this is r+.

**Comment by jotes** [comment]
@mathjazz could you look again?

File: app.json [hunk 5]
```diff
@@ -32,6 +32,10 @@
      }
    ],
    "env": {
+     "SITE_URL": {
+       "description": "Access url to the instance, has to be https://.herokuapp.com without a trailing slash.",
```
Comment (mathjazz) [inline app.json]: `"Base URL of the site. Has to be https://{app-name}.herokuapp.com."`

Comment (jotes) [reply app.json]: done :-)

**Pull request merged**